coordinator/src/lib.rs

```diff
@@ -48,7 +48,7 @@ pub struct Builder<C: context::Context> {
 }
 
 impl<C: context::Context> Builder<C> {
-    fn new<CTX: context::Context>(ctx: CTX) -> Builder<CTX> {
```

Comment:
I think clippy said you that `new` function should return `Self` type, doesn't it?
You can make the function to return Self. The `Builder` object already has the type parameter `C`, you don't need to create a new type with a new type parameter.

Reply:
@Byeongjee Please check this comment

Reply:
you mean that we should erase the type parameter CTX?
Then I agree with you.